core/src/services/gcs/writer.rs

```diff
@@ -163,9 +163,26 @@ impl oio::Write for GcsWriter {
         }
     }
 
-    // TODO: we can cancel the upload by sending a DELETE request to the location
     async fn abort(&mut self) -> Result<()> {
-        Ok(())
+        let location = if let Some(location) = &self.location {
+            location
+        } else {
+            return Ok(());
+        };
+
+        let resp = self.core.gcs_abort_resumable_upload(location).await?;
+
+        match resp.status() {
```

Comment:
How about using `resp.status().as_u16()` so that we don't need to `unwrap()`?

Reply:
We can write like:

```rust
match resp.status().as_u16() {
  499 => xxx
  _ => xxx
}
```

Reply:
"I prefer to maintain consistency with the most beautiful code instead.

Reply:
This doesn't seem more beautiful either. 🤣 
Updated.

Reply:
> This doesn't seem more beautiful either.

Feel free to improve it.